### PR TITLE
Iohk nix

### PR DESCRIPTION
**Pull request by disassembler**
## Description

This is a complete rewrite of all the nix code in cardano-sl. Even though we don't anticipate having another release of SL, this will ease changes needed for the byron-proxy until we're able to shutdown the proxies.

[hydra jobs](https://hydra.iohk.io/jobset/Cardano/cardano-sl-pr-4183#tabs-jobs)

Task List:
- [x] add iohk-nix
- [x] switch to only building with nix-tools
- [x] cleanup of default.nix and release.nix
- [x] move launch scripts to nix directory
- [x] move tests to nix directory
- [x] fix daedalus bridge
- [x] add proposalGui to connectScripts
- [x] get validateYamlJson working again
- [x] add all connect scripts and docker images to required jobset
- [x] mapTestOn connectScripts to build for multiple platforms
- [ ] test daedalus bridge on all 3 platforms
- [ ] test wallet and explorer connectScripts on linux/mac
- [ ] test docker containers
- [ ] add all test derivations to required jobset
- [ ] fix hard-coded version/rev propagation

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## How to merge

Send the message `bors r+` to merge this PR. For more information, see
[`docs/how-to/bors.md`](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/how-to/bors.md).

### CODE REVIEW

**Comment by angerman** [comment]
See input-output-hk/iohk-nix#126 for the clock issue. The rest is mostly disabling haddock and forcing to ignore package bounds, which stackage probably does anyway.

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 steps:
-  - label: 'stack2nix'
-    command: 'scripts/check-stack2nix.sh'
+  - label: Check Hydra evaluation of release.nix
+    command: 'nix-build -A check-hydra lib.nix -o check-hydra.sh && ./check-hydra.sh'
     agents:
       system: x86_64-linux
 
-  - label: 'release.nix'
-    command: 'scripts/ci/check-hydra.sh'
```
Comment (cleverca22) [inline .buildkite/pipeline.yml]: is the old `check-hydra.sh` also being deleted?

File: lib.nix [hunk 2]
```diff
@@ -93,27 +68,4 @@ in lib // (rec {
     (name: env: f (env // { environment = name; }))
     environments;
 
-  runHaskell = name: hspkgs: deps: env: code: let
```
Comment (cleverca22) [inline lib.nix]: this was being used by the yaml validation tests in hydra, are those still being ran?

**Pull request merged**